packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -558,6 +595,85 @@ class StdoutLogger extends Logger {
   }
 }
 
+typedef _Writter = void Function(String message);
+
+/// Wraps the message in a box, and writes the bytes by calling [write].
+///
+///  Example output:
+///
+///   ┌─ [title] ─┐
+///   │ [message] │
+///   └───────────┘
+///
+/// When [title] is provided, the box will have a title above it.
+///
+/// The box width never exceeds [wrapColumn].
+///
+/// If [wrapColumn] is not provided, the default value is 100.
+void _generateBox({
+  required String message,
+  required int wrapColumn,
+  required _Writter write,
+  required Terminal terminal,
+  String? title,
+}) {
+  const int kPaddingLeftRight = 1;
+  const int kEdges = 2;
+
+  final int maxTextWidthPerLine = wrapColumn - kEdges - kPaddingLeftRight * 2;
+  final List<String> lines = wrapText(message, shouldWrap: true, columnWidth: maxTextWidthPerLine).split('\n');
+  final List<int> lineWidth = lines.map((String line) => _getColumnSize(line)).toList();
+  final int maxColumnSize = lineWidth.reduce((int currLen, int maxLen) => max(currLen, maxLen));
+  final int textWidth = min(maxColumnSize, maxTextWidthPerLine);
+  final int textWithPaddingWidth = textWidth + kPaddingLeftRight * 2;
+
+  write('\n');
+
+  // Write `┌─ [title] ─┐`.
+  write('┌');
+  write('─');
+  if (title == null) {
+    write('─' * (textWithPaddingWidth - 1));
+  } else {
+    write(' ${terminal.bolden(title)} ');
+    write('─' * (textWithPaddingWidth - title.length - 3));
+  }
+  write('┐');
+  write('\n');
+  write('│');
+  write(' ' * textWithPaddingWidth);
+  write('│');
+  write('\n');
+
+  // Write `│ [message] │`.
+  for (int lineIdx = 0; lineIdx < lines.length; lineIdx++) {
+    write('│');
+    write(' ' * kPaddingLeftRight);
+    write(lines[lineIdx]);
+    final int remainingSpacesToEnd = textWidth - lineWidth[lineIdx];
+    write(' ' * (remainingSpacesToEnd + kPaddingLeftRight));
+    write('│');
+    write('\n');
+  }
+
+  // Write `└───────────┘`.
+  write('│');
+  write(' ' * textWithPaddingWidth);
+  write('│');
+  write('\n');
+  write('└');
+  write('─' * textWithPaddingWidth);
+  write('┘');
+  write('\n');
+}
+
+final RegExp _ansiEscapePattern = RegExp(r'(\u001B|\u001b)\[([0-9][a-zA-Z]|[a-zA-Z])');
```

Comment:
isn't `\u001B|\u001b` redundant?

Reply:
in any case i think this is the wrong pattern for matching escape sequences, according to wikipedia it should be something like `'\x1B\\[[\x30-\x3F]*[\x20-\x2F]*[\x40-\x7E]'` (see https://en.wikipedia.org/wiki/ANSI_escape_code) (note this is _not_ a raw string, the escapes are actual string escapes not regexp escapes)

Reply:
oh nice. TIL - hex ranges for character matching. 
I only looked at the currently supported escape characters in the tool, but that makes a ton of sense.